src/Credentials/CredentialProvider.php

```diff
@@ -406,6 +442,58 @@ public static function process($profile = null, $filename = null)
         };
     }
 
+    /**
+     * Assumes role for profile that includes role_arn
+     *
+     * @return callable
+     */
+    private static function loadRoleProfile($profiles, $profileName, $filename, $stsClient)
+    {
+        $roleProfile = $profiles[$profileName];
+        $externalId = isset($roleProfile['external_id']) ? $roleProfile['external_id'] : '';
+        $mfaSerial = isset($roleProfile['mfa_serial']) ? $roleProfile['mfa_serial'] : '';
+        $roleArn = isset($roleProfile['role_arn']) ? $roleProfile['role_arn'] : '';
+        $roleSessionName = isset($roleProfile['role_session_name']) ? $roleProfile['role_session_name'] : 'aws-sdk-php-' . date('Uv');
+
+        if (!isset($profiles[$profileName]['source_profile'])) {
+            return self::reject("source_profile is not set using profile " .
+                $profileName
+            );
+        }
+
+        $sourceProfileName = $roleProfile['source_profile'];
+        if (!isset($profiles[$sourceProfileName]))
```

Comment:
`empty()` probably better here.

Reply:
I think it's useful as-is.  If a source_profile isn't set, this raises the error that it should be set.

If the profile set is empty (or missing credentials), that will get raised elsewhere as ```No credentials present in INI profile```